cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -256,6 +256,23 @@ Status ByteArrayStatisticsAsScalars(const Statistics& statistics,
   return Status::OK();
 }
 
+Result<std::shared_ptr<ChunkedArray>> ViewOrCastChunkedArray(
+    const std::shared_ptr<ChunkedArray>& array,
+    const std::shared_ptr<DataType>& logical_value_type) {
+  auto view_result = array->View(logical_value_type);
+  if (view_result.ok()) {
+    return view_result.ValueOrDie();
+  } else {
+    ::arrow::ArrayVector casted_chunks;
+    for (const auto& chunk : array->chunks()) {
+      ARROW_ASSIGN_OR_RAISE(auto cast_chunk,
+                            ::arrow::compute::Cast(chunk, logical_value_type));
+      casted_chunks.push_back(cast_chunk.make_array());
+    }
```

Comment:
Two comments here:
1. There is a `Cast` function overload that accepts a `Datum`, why not use that instead of casting each chunk individually?
2. We should pass the right `MemoryPool` when casting.

Reply:
This makes sense; I didn't know about (1). Maybe a dumb question for (2), but how should I determine which `MemoryPool` to use?